File: Find the maximum GCD of the siblings of a Binary Tree.cpp
class Solution {
    int ans, g;

    void solve(Node *root)
    {
        if(root==NULL)
        return;
        if(root->left!=NULL && root->right!=NULL)
        {
            int h=__gcd(root->left->data,root->right->data);
            if(h>g)
            {
                g=h;
                ans=root->data;
            }
            else if(h==g)
            {
                ans=max(ans,root->data);
            }
        }
        solve(root->left);
        solve(root->right);
    }

public:
    int maxGCD(Node* root) {
        ans = 0;
        g = 0;
        solve(root);
        return ans;
    }
};

File: Find the maximum GCD of the siblings of a Binary Tree.py
from math import gcd
class Solution:
    def maxGCD(self, root):
        # code here
        return self.solve(root)[1]
    
    def solve(self,node):
            if node is None:
                return (0,0)
            x = (0,0)
            if node.left and node.right:
                x = (gcd(node.left.data, node.right.data),node.data)
            y = self.solve(node.left)
            z = self.solve(node.right)
            return max(x,y,z)
